src/testers.rs

```diff
@@ -58,63 +62,52 @@ pub fn undefined(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraRe
 /// Returns true if `value` is a string. Otherwise, returns false.
 pub fn string(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
     try!(number_args_allowed(name, "string", 0, params.len()));
+    try!(value_defined(name, "string", &value));
 
-    match value {
-        Some(Value::String(_)) => Ok(true),
-        None => value_is_undefined(name, "string"),
+    match value.unwrap() {
```

Comment:
I disagree on that, if the unwrap is safe (like here) I don't mind it. That or `expect`. Test(s) cover the case where the line above is removed.

I'll change it though, the `and_then` approach is neater